dagster-io/dagster
# [dagster-io/ui] Declaration files for TS, Rollup for built JS

## Summary

Update @dagster-io/ui:

- Use Rollup to generate a JS bundle
- Create declaration files for TS

`ui` package:

- Use Rollup to generate JS bundles for use by consumers.
  - Split some components into their own bundles. The goal here is to drastically reduce the overhead for consumers that just need a button or something small. We can add to this list as needs arise.
- Remove direct imports of blueprint and font CSS. These bloat the final bundle, and aren't actually needed here by consumers: some consumers won't need any components with Blueprint styling, and others can import fonts via Google Fonts API.
- Create `GlobalInter` and `GlobalInconsolata`, modules that create global styles with base64 fonts. These can be rendered alongside other global styles at the app root.
- Create a changelog. As noted in the Readme, this lib isn't intended or supported for uses outside of our own repositories, but we should still track changes for internal users.

`core` and `app` packages:

- Create `blueprint.css` for `AppProvider` and the `core` Storybook. This imports the relevant Blueprint CSS.
- In TypeScript, point `@dagster-io/ui` path directly at `src` since we can do this instead of relying on the built type declarations.

## Test Plan

Load `dagit` and `dagit-cloud`. Verify that the apps build correctly and render/behave successfully in dev.

Create a `test-ui` CRA locally. Add `@dagster-io/ui` as a linked dependency to my local version.

- Verify that components are imported and typechecked.
- Run the app, verify that components render properly.
- Run a prod build. Verify that the build doesn't blow up due to importing `Box`, which is a split bundle in the UI package.
